### ISSUE TITLE
Unsupported Scan

### ISSUE THREAD

**Issue by nh2** [sorintlab/stolon#329]
Hi, I did some invocations of `stolonctl update`, and after some time ended up with this in the logs:

```
keeper.go:526: cannot get configured pg parameters error=sql: Scan error on column index 0: unsupported Scan, storing driver.Value type <nil> into type *string
```

What does that mean?

**Comment by sgotti** [comment]
@nh2 looks like the query on the pg parameters returned a value with a null value. We should handle such case because it'll cause the keeper to be marked as failed but I never had this error. Probably you updated some pgparameters and one of those caused this. Can you provide which parameters (or the clusterspec) you changed so I can try to reproduce it? What is the postgres version? Thanks.

**Comment by nh2** [comment]
@sgotti I haven't recorded it fully because I was just playing around on some temp machines, but think what I did was this:

First

```
stolonctl update --patch '{ "pgParameters": { "log_statement": "all", "log_min_duration_statement": "0" } }'
```

To turn on some logs.

Then `stolonctl update` without `--patch` with a full spect that didn't have `pgParameters`.

On postgres 9.6.

**Comment by sgotti** [comment]
@nh2 Thanks. I reproduces this using your steps. Basically if you remove some parameters that requires a server restart from the postgresql.conf the pg_file_settings view will contain a row for every of these removed parameter with name and setting value set to null and the error set to the cause:

```
postgres=# select * from pg_file_settings;                                                                                                                                       
sourcefile                        | sourceline | seqno |           name            |       setting       | applied |                                         error                                             
----------------------------------------------------------------------------------------------+------------+-------+---------------------------+---------------------+---------+-------------------------------
 /stolon/postgres/postgresql.conf |          1 |     1 | wal_keep_segments         | 8                   | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          2 |     2 | max_wal_senders           | 27                  | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          3 |     3 | synchronous_standby_names |                     | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          4 |     4 | listen_addresses          | 127.0.0.1,localhost | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          5 |     5 | port                      | 5434                | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          6 |     6 | max_replication_slots     | 20                  | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          7 |     7 | wal_level                 | hot_standby         | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          8 |     8 | hot_standby               | on                  | t       |                                                                                           
 /stolon/postgres/postgresql.conf |          9 |     9 | unix_socket_directories   | /tmp                | t       |                                                                                           
                                  |            |    10 |                           |                     | f       | parameter "dynamic_shared_memory_type" cannot be changed without restarting the server    
                                  |            |    11 |                           |                     | f       | parameter "logging_collector" cannot be changed without restarting the server             
                                  |            |    12 |                           |                     | f       | parameter "max_connections" cannot be changed without restarting the server               
                                  |            |    13 |                           |                     | f       | parameter "shared_buffers" cannot be changed without restarting the server                

```
since the scanned values aren't pointers a null value will return such error. I create #332 to fix this.

**Comment by sgotti** [comment]
Fixed by #332
